src/neuroconv/tools/testing/mock_ttl_signals.py

```diff
@@ -0,0 +1,290 @@
+    channel_noise_int16_default = 15
+    default_gain_to_volts = 152.58789062 * 1e-6
+
+    if np.issubdtype(dtype, np.unsignedinteger):
+        # If data type is an unsigned integer, increment the signed default values by the midpoint of the unsigned range
+        shift = np.floor(np.iinfo(dtype).max / 2)
+        baseline_mean_int16_default += shift
+        signal_mean_int16_default += shift
+
+    if np.issubdtype(dtype, np.integer):
+        baseline_mean = baseline_mean or baseline_mean_int16_default
+        signal_mean = signal_mean or signal_mean_int16_default
+        channel_noise = channel_noise or channel_noise_int16_default
+
+        assert np.issubdtype(type(baseline_mean), np.integer), (
+            "If specifying the 'baseline_mean' manually, please ensure it matches the 'dtype'! "
+            f"Received '{type(baseline_mean).__name__}', should be an integer."
+        )
+        assert np.issubdtype(type(signal_mean), np.integer), (
+            "If specifying the 'signal_mean' manually, please ensure it matches the 'dtype'! "
+            f"Received '{type(signal_mean).__name__}', should be an integer."
+        )
+        assert np.issubdtype(type(channel_noise), np.integer), (
+            "If specifying the 'channel_noise' manually, please ensure it matches the 'dtype'! "
+            f"Received '{type(channel_noise).__name__}', should be an integer."
+        )
+    else:
+        baseline_mean = baseline_mean or baseline_mean_int16_default * default_gain_to_volts
+        signal_mean = signal_mean or signal_mean_int16_default * default_gain_to_volts
+        channel_noise = channel_noise or channel_noise_int16_default * default_gain_to_volts
+
+        assert np.issubdtype(type(baseline_mean), np.floating), (
+            "If specifying the 'baseline_mean' manually, please ensure it matches the 'dtype'! "
+            f"Received '{type(baseline_mean).__name__}', should be a float."
+        )
+        assert np.issubdtype(type(signal_mean), np.floating), (
+            "If specifying the 'signal_mean' manually, please ensure it matches the 'dtype'! "
+            f"Received '{type(signal_mean).__name__}', should be a float."
+        )
+        assert np.issubdtype(type(channel_noise), np.floating), (
+            "If specifying the 'channel_noise' manually, please ensure it matches the 'dtype'! "
+            f"Received '{type(channel_noise).__name__}', should be a float."
+        )
+
+    np.random.seed(seed=random_seed)
+    num_frames = np.ceil(signal_duration * sampling_frequency_hz).astype(int)
+    trace = (np.random.randn(num_frames) * channel_noise + baseline_mean).astype(dtype)
+
+    if ttl_times is not None:
+        ttl_times = np.array(ttl_times)
+    else:
+        # Start halfway through an off pulse
+        total_cycle_duration = ttl_off_duration + ttl_on_duration
+        cycle_start_times = np.arange(0, signal_duration, total_cycle_duration)
+        num_cycles = len(cycle_start_times)
```

Comment:
I feel like it would be simpler to have the user input the total_cycle_duration or the num_cycles than the ttl_off_duration but it's fine to do it this way I guess

Reply:
Yeah, I refactored the arguments and design of this now to be a lot more intuitive and less magical.

The default is still a regular 1 second off, 1 second on pulse over the `signal_duration`, but instead of mixing a bunch of inter-conditional arguments, there's just the `signal_duration` (total length), the `ttl_times`, and the `ttl_duration` (how long to stay in 'on' state when triggered by one of the `ttl_times`).

As such, it's the user has full control over the pattern of TTL times, and can enforce them to be regular simply by specifying regular times and vice versa for irregular.

Reply:
This also makes the tests in [Sync II] much easier to read since you can automatically tell how many TTL pulses there are and roughly how they ought to line up over time